src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import I18n from 'i18n-js';
+import PropTypes from 'prop-types';
+import Events from '../../../utils/helpers/events';
+import DateHelper from '../../../utils/helpers/date';
+import DateValidator from '../../../utils/validations/date';
+import tagComponent from '../../../utils/helpers/tags';
+import DatePicker from './date-picker.component';
+import InputDecoratorBridge from '../input-decorator-bridge';
+import StyledDateInput from './date.style';
+import Textbox from '../textbox';
+
+const isoDateFormat = 'YYYY-MM-DD';
+const today = DateHelper.todayFormatted(isoDateFormat);
+
+/**
+ * A Date widget.
+ *
+ * == How to use a Date in a component:
+ *
+ * In your file
+ *
+ *   import Date from 'carbon-react/lib/components/Date';
+ *
+ * To render the Date:
+ *
+ *   <Date name="myDate" />
+ *
+ * @class Date
+ * @constructor
+ * @decorators {Input,InputIcon,InputLabel,InputValidation}
+ */
+class Date extends React.Component {
+  static propTypes = {
+    ...Textbox.propTypes,
+    /** Automatically focus on component mount */
+    autoFocus: PropTypes.bool,
+    /** Used to provide additional validations on composed components */
+    internalValidations: PropTypes.array,
+    /** Minimum possible date YYYY-MM-DD */
+    minDate: PropTypes.string,
+    /** Maximum possible date YYYY-MM-DD */
+    maxDate: PropTypes.string,
+    /** Specify a callback triggered on blur */
+    onBlur: PropTypes.func,
+    /** Specify a callback triggered on change */
+    onChange: PropTypes.func,
+    /** Specify a callback triggered on focus */
+    onFocus: PropTypes.func,
+    /** The current date */
+    value: PropTypes.string
+  };
+
+  static defaultProps = {
+    value: today,
+    internalValidations: [new DateValidator()]
+  };
+
+  // Stores the document - allows us to override it different contexts, such as when running tests.
+  _document = document;
+
+  bridge = React.createRef() // this is a reference to the input decorator bridge component
+
+  blurBlocked = false; // stops the blur callback from triggering (closing the list) when we don't want it to
+
+  state = {
+    isDatePickerOpen: false,
```

Comment:
Should be `datePickerIsOpen`.

Reply:
Do we have a name convention for that? I'm deliberately placing `is` or `has` as a prefix of the name to indicate that the value is a boolean.

Reply:
Well it's not a convention really, it's the way it reads in plain english.

`isDatePickerOpen` suggests that the value will be a function:
```js
if( isDatePickerOpen ) { } // not good grammar

if( datePickerIsOpen ) { } // good grammar
```

Reply:
I think that we should put more focus on code readability than the english gramatics. When you're reviewing code `isDatePickerOpen` on the first sight is more suggestive that's a boolean than `datePickerIsOpen`. 
If `isDatePickerOpen` would be a function (assuming that it returns a boolean and it has no side effects), then we could treat it as boolean and we'll not have to think about what that method is doing.

Reply:
I commented on it because as I read the code I didn't think the term had good code readability, because when you read it on its own it looks like a function (which it isn't) and when you read it in the context of an `if` statement it doesn't read like English.

Reply:
A compromise might be to leave out the `is` entirely, as `datePickerOpen` is more suggestive of a bool, and `if( datePickerOpen )` is more idiomatically English.

Reply:
It's a common convention to prefix booleans with `is` or `has`, I have used it in my previous projects and a lot of people is using that convention - for example the Enzyme team. 

As for the leaving out the `is` - `datePickerOpen` could be a string and it will be valid in a conditional.

Reply:
I think it is conventional for method names and attributes on an instance: `person.isAllve` or `expect(foo).toHaveBeenCalled()` are reasonable, `toHaveBeenCalled(foo)` is less reasonable. 

I don't see how `datePickerOpen` could imply a string type.

Reply:
`isOpen` would suggest that the `Date` component have a state open, but this state is responsible for rendering of the `DatePicker` component.

Reply:
It can be helpful https://wiki.eclipse.org/Recommenders/CodingConventions#is_prefix_should_be_used_for_boolean_variables_and_methods.